storybookjs/storybook
# React-Docgen: Make error-handling more gentle

Closes https://github.com/storybookjs/storybook/issues/25053





## What I did

 Storybook 7.6, we have replaced the babel-plugin-react-docgen plugin with a custom Webpack5 loader, which works independently of the used compiler (SWC, Babel, ...). The loader throws an Error, though, as soon as a file cannot be processed.
 
An Error is not thrown anymore but instead log a warning to the user. In --loglevel=debug mode a stack trace should be available to report bug reports for react-docgen.
 
## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Clone https://github.com/zooniverse/front-end-monorepo
2. Checkout dependabot/npm_and_yarn/storybook/react-webpack5-7.6.1
3. `yarn install` (Node 20)
4. Go to `packages/lib-classifier` and run `yarn install`
5. `yarn storybook` -> it should fail
6. Precompile `preset-react-webpack` from the monorepo and swap out the files
7. A warning should be shown
8. Start Storybook with `yarn storybook loglevel=debug` -> Now a stacktrace is shown as well

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
